qiskit/transpiler/passes/optimization/commutative_optimization.py

```diff
@@ -0,0 +1,70 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2025.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Commutative Optimization transpiler pass."""
+
+
+from qiskit.transpiler.basepasses import TransformationPass
+from qiskit.circuit.commutation_library import SessionCommutationChecker as scc
+
+from qiskit._accelerate import commutative_optimization
+
+from qiskit.transpiler.passes.utils.control_flow import trivial_recurse
+
+
+class CommutativeOptimization(TransformationPass):
+    """
+    Cancel/merge gates exploiting commutativity relations.
+
+    The pass will:
+
+    * Cancel pairs of inverse gates.
+    * Cancels pairs of gates that are inverse up to a global phase (adjusting
+      the global phase accordingly).
+    * Combines different types of RZ-rotations into an RZ-gate.
```

Comment:
It also merges Pauli rotations 🙂 Maybe you can keep it generic and just say "Attempt to merge consecutive gates, including Pauli rotations" or sth the likes